collections/distinct_by_test.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "https://deno.land/std@0.100.0/testing/asserts.ts";
+import { distinctBy } from "./distinct_by.ts";
+
+function distinctByTest<I>(
+  array: Array<I>,
+  selector: (element: I) => unknown,
+  expected: Array<I>,
+  message?: string,
+) {
+  const actual = distinctBy(array, selector);
+  assertEquals(actual, expected, message);
+}
+
+Deno.test({
+  name: "[collections/distinctBy] identities on empty array",
+  fn() {
+    distinctByTest(
+      [],
+      () => {},
+      [],
+    );
+  },
+});
+
+Deno.test({
+  name: "[collections/distinctBy] gets head on noop selector",
```

Comment:
As `() => {}` is equivalent to `it => undefined` here, it would feel like unexpected magic behavior to me if it would explicitly ignore equality (distinctity? is that a word?) for `undefined`. What about `null`? `""`? What if `undefined` is a legit value in my model?

I wrote those tests before implementing the function, so to mee it seemed like the intuitive behavior to just distinct all values by their produced value - no matter what it is.